lib/array_intersection.rb

```diff
@@ -1,6 +1,22 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*m), where n is the number of elements in array1 and m is the number of elements in array2
+# Space complexity: O(n), where n is less than m
 def intersection(array1, array2)
-  raise NotImplementedError
+  return [] if array1 == nil || array2 == nil
```

Comment:
nil might be more appropriate if one of the input arrays is nil. The problem description doesn't specify, so your choice is still valid.